docs/docs/kb/03-config.md

```diff
@@ -4,187 +4,255 @@ description: Primary configuration file to describe the development environment
-| mnemonic | N        | String          | Mnemonic used to generate an account. This field is ignored if `address` is specified.                                          |
-
-Note that you can only use `address` OR `mnemonic` for an account. You can't use both, because an address is derived
-from a mnemonic.
-
-If an account is a validator account (`alice` by default), it cannot have an `address` field.
-
-**accounts example**
-
-```yaml
+```yml
 accounts:
   - name: alice
-    coins: [ "1000token", "100000000stake" ]
+    coins: ['20000token', '200000000stake']
   - name: bob
-    coins: [ "500token" ]
-    address: cosmos1adn9gxjmrc3hrsdx5zpc9sj2ra7kgqkmphf8yw
+    coins: ['10000token', '100000000stake']
 ```
 
-## build
-
-| Key      | Required | Type             | Description                                                                                                  |
-|----------|----------|------------------|--------------------------------------------------------------------------------------------------------------|
-| main     | N        | String           | When an app contains more than one main Go package, required to define the path of the chain's main package. |
-| binary   | N        | String           | Name of the node binary that is built, typically ends with `d`.                                              |
-| ldflags  | N        | List of Strings  | ldflags to set version information for go applications.                                                      |
+Ignite uses information from `accounts` when initializing the chain with `ignite
+chain init` and `ignite chain start`. In the example above Ignite will add two
+accounts to the `genesis.json` file of the chain.
+
+`name` is a local name of a key pair associated with an account. Once the chain
+is initialized and started, you will be able to use `name` when signing
+transactions. With the configuration above, you'd be able to sign transactions
+both with Alice's and Bob's accounts like so `exampled tx bank send ... --from
+alice`.
+
+`coins` is a list of token balances for the account. If a token denomination is
+in this list, it will exist in the genesis balance and will be a valid token.
+When initialized with the config file above, a chain will only have two accounts
+at genesis (Alice and Bob) and two native tokens (with denominations `token` and
+`stake`).
+
+By default, every time a chain is re-initialized, Ignite will create a new key
+pair for each account. So even though the account name can remain the same
+(`bob`), every chain reinitialize it will have a different mnemonic and address.
+
+If you want an account to have a specific address, provide the `address` field
+with a valid bech32 address. The prefix (by default, `cosmos`) should match the
+one expected by your chain. When an account is provided with an `address` a key
+pair will not be generated, because it's impossible to derive a key from an
+address.
```

Comment:
Very interesting to add those details, but in that case, where's the key pair supposed to be ?

Reply:
https://github.com/ignite/cli/pull/3042/commits/fe4cc713bbdc66ba7e2df8605b73247c560a8e10

Reply:
Thanks for adding the consequences of that, but then, what is the usecase ?

My understanding of that `address` field is to use a pre-existing key pair, generated outside of ignite. So if i'm not wrong, maybe we should mention it ?

Reply:
https://github.com/ignite/cli/pull/3042/commits/5e0e2412ee45d7b3cd4c1df4c522251022197143